### ISSUE TITLE
Malloc returned values doesn't checked

### ISSUE THREAD

**Issue by NoamRabi20** [facebook/rocksdb#13099]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior
Malloc return values should be verified, and in case of unsuccessful malloc, error should be propagate

### Actual behavior
unsuccessful malloc will return null, and the following action will cause crash

### Steps to reproduce the behavior
run on low memory system 

###examples
db/c.cc (7 instances)
cache/lru.cache.cc (2 instances)

**Comment by gitlijian** [comment]
hi， @NoamRabi20 Is this the way to modify it? If possible, I will perform exception detection at each call to malloc.
![image](https://github.com/user-attachments/assets/89205afa-807e-4fa7-970b-e61e091825be)

**Comment by NoamRabi20** [comment]
yes, this is good approach, just need to make sure at each call how the code handle errors (return null in this specific case) it may be different between calls

**Comment by DigraJatin** [comment]
Looks like this is not fixed, i can take this
